core/upgrade/src/main/java/org/opennms/upgrade/implementations/EventParameterMigratorOffline.java

```diff
@@ -0,0 +1,159 @@
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.upgrade.implementations;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.Map;
+
+import org.opennms.core.db.DataSourceFactory;
+import org.opennms.netmgt.events.api.EventParameterUtils;
+import org.opennms.netmgt.xml.event.Parm;
+import org.opennms.upgrade.api.AbstractOnmsUpgrade;
+import org.opennms.upgrade.api.OnmsUpgradeException;
+
+public class EventParameterMigratorOffline extends AbstractOnmsUpgrade {
+
+    public EventParameterMigratorOffline() throws OnmsUpgradeException {
+        super();
+    }
+
+    @Override
+    public int getOrder() {
+        return 8;
+    }
+
+    @Override
+    public String getDescription() {
+        return "Moves event parameters from eventparms column to table";
+    }
+
+    @Override
+    public boolean requiresOnmsRunning() {
+        return false;
+    }
+
+    @Override
+    public void preExecute() throws OnmsUpgradeException {
+        try (final Connection connection = DataSourceFactory.getInstance().getConnection()) {
+            final Statement preExecutionStatement = connection.createStatement();
+            try (final ResultSet preExecutionResultSet = preExecutionStatement.executeQuery("SELECT EXISTS (SELECT 1 FROM information_schema.columns WHERE table_schema='public' AND table_name='events' AND column_name='eventparms')")) {
+                preExecutionResultSet.next();
+                if (!preExecutionResultSet.getBoolean(1)) {
+                    throw new OnmsUpgradeException("The column 'eventParms' does not exists anymore");
+                }
+            } catch (SQLException e) {
+                e.printStackTrace();
+                throw new OnmsUpgradeException("Error checking for column 'eventParms'", e);
+            }
+        } catch (SQLException e) {
+            throw new OnmsUpgradeException("Error opening database connection", e);
+        }
+    }
+
+    @Override
+    public void postExecute() throws OnmsUpgradeException {
+    }
+
+    @Override
+    public void rollback() throws OnmsUpgradeException {
+    }
+
+    @Override
+    public void execute() throws OnmsUpgradeException {
+        long eventCount = 0, parameterCount = 0;
+
+        try (final Connection connection = DataSourceFactory.getInstance().getConnection()) {
+
+            final Statement preMigrationStatement = connection.createStatement();
+            final ResultSet preMigrationResultSet = preMigrationStatement.executeQuery("SELECT EXISTS (SELECT 1 FROM information_schema.columns WHERE table_schema='public' AND table_name='events' AND column_name='eventparms')");
```

Comment:
At this point I can also use the ResultSet's meta data of a "SELECT * FROM events LIMIT 0" statement to look for the "eventParms" column. I'm also using this information_schema query in the liquibase's changelog.xml to drop the columns if all eventParms are NULL.